frontend/packages/console-shared/src/components/pod/PodRingDataController.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { Firehose, FirehoseResource } from '@console/internal/components/utils';
+import { Pod, Resource } from '../../types/pod';
+import { transformPodRingData } from './podring-utils';
+
+export interface PodRingDataResources {
+  replicationControllers: Resource;
+  pods: Resource;
+  replicasets: Resource;
+  DeploymentConfig: Resource;
+  Deployment: Resource;
+}
+
+export interface RenderPropsType {
+  loaded: boolean;
+  loadError: any;
+  data: {
+    [name: string]: {
+      pods: Pod[];
+    };
+  };
+}
+
+export interface ControllerProps {
+  loaded?: boolean;
+  loadError?: any;
+  resources?: PodRingDataResources;
+  kind: string;
+  render(RenderProps: RenderPropsType): React.ReactElement;
+}
+
+export interface PodRingDataControllerProps {
+  namespace: string;
+  kind: string;
+  render(RenderProps: RenderPropsType): React.ReactElement;
+}
+
+const Controller: React.FC<ControllerProps> = React.memo(
+  ({ resources, render, loaded, loadError, kind }) => {
+    return render({
+      loaded,
+      loadError,
+      data: loaded ? transformPodRingData(resources, kind) : null,
+    });
+  },
+);
+
+const PodRingDataContoller: React.FC<PodRingDataControllerProps> = ({
+  namespace,
+  kind,
+  render,
+}) => {
+  let resources: FirehoseResource[] = [
+    {
+      isList: true,
+      kind: 'Pod',
+      namespace,
+      prop: 'pods',
+    },
+    {
+      isList: true,
+      kind: 'ReplicaSet',
+      namespace,
+      prop: 'replicasets',
+    },
+    {
+      isList: true,
+      kind: 'ReplicationController',
+      namespace,
+      prop: 'replicationControllers',
+    },
+  ];
+
+  switch (kind) {
```

Comment:
@christianvogt Which model are you talking about?

Reply:
@vikram-raj instead of passing `kind` in `PodRingDataContoller` if you pass whole dc object, i think can avoid this `switch`. WDYT?

```
{
      isList: true,
      kind: dc.kind,
      namespace,
      prop: dc.plural,
}
```

Reply:
@invincibleJai Here dc of type ```K8sResourceKind``` which have no plural property.

Reply:
ah , can dc have something other than deployment and deploymentconfig ?

Reply:
@invincibleJai No, currently it will be either deployment or deploymentConfig.